pipeline/test/services/service-cluster/testOpensearch.sh

```diff
@@ -0,0 +1,385 @@
+    echo -e "failure ❌";
+    echo "$cluster_health" | jq
+  else
+    echo -e "success ✔";
+  fi
+}
+
+check_opensearch_snapshots_status(){
+  echo -ne "Checking opensearch snapshots status ... "
+  repo_name=$(yq4 -e '.opensearch.snapshot.repository' "${config['config_file_sc']}")
+  repo_exists_status=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_snapshot/${repo_name}" | jq "select(.error)")
+  if [[ -z "$repo_exists_status" ]]; then
+    if kubectl get "cronjob" -n "opensearch-system" "opensearch-backup" &> /dev/null
+    then
+        last_schedule=$(kubectl get cronjob -n opensearch-system opensearch-backup -ojsonpath='{.status.lastScheduleTime}')
+        if [[ $last_schedule != $(date +"%Y-%m-%dT12:00:00Z") ]]; then
+          echo -e "failure ❌";
+          echo "[ERROR] opensearch-backup cronjob has not been scheduled"
+        fi
+        echo -e "success ✔";
+        return 0
+    else
+        echo -e "failure ❌";
+        echo "[ERROR] opensearch-backup cronjob doesn't exist"
+    fi
+  else
+    echo -ne "failure ❌";
+    echo -e "$repo_exists_status" | jq
+  fi
+}
+
+check_opensearch_indices(){
+  echo -ne "Checking opensearch indices status ... "
+  debug_msg=""
+  no_error=true
+
+  for index in 'other' 'kubernetes' 'kubeaudit' 'authlog'; do
+    res=$(curl -w "%{http_code}"  -o /dev/null -ksIL -u admin:"${adminPassword}" -X HEAD "https://opensearch.${opsDomain}/${index}")
+    if [[ $res != "200" ]]; then
+      debug_msg+="[ERROR] Missing index : ${index}\n"
+      no_error=false
+    fi
+  done
+
+  if $no_error; then
+    echo "success ✔"
+    echo -e "[DEBUG] All indices are present"
+  else
+      echo "failure ❌";
+      echo -e "$debug_msg"
+  fi
+}
+
+check_opensearch_breakers(){
+  echo -ne "Checking opensearch breakers ... "
+  data=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_nodes/_all/stats/breaker")
+  no_error=true
+  debug_msg=""
+  nodes_data=$(echo "$data" | jq ".nodes")
+  readarray -t nodes < <(jq -c 'to_entries[]' <<< "$nodes_data")
+
+  for node in "${nodes[@]}"; do
+    node_name=$(jq '.key' <<< "$node")
+    readarray -t breakers < <(jq -c '.value.breakers | to_entries[]' <<< "$node")
+    for breaker in "${breakers[@]}"; do
+      tripped=$(jq ".value.tripped" <<< "$breaker")
+      name=$(jq ".key" <<< "$breaker")
+      if [[ $tripped == "1" ]]; then
+        no_error=false
+        debug_msg+="[DEBUG] A circuit breaker : $name has been triggered for node: $node_name\n"
+      fi
+    done
+  done
+
+  if $no_error; then
+    echo "success ✔"
+    echo "[DEBUG] None of the circuit breakers has been triggered"
+  else
+    echo "failure ❌";
+    echo -e "$debug_msg"
+  fi
+
+}
+
+check_opensearch_aliases(){
+  echo -ne "Checking opensearch aliase indices mapping ... "
+  no_error=true
+  debug_msg=""
+
+  curl -sk -o /tmp/response -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_cat/aliases"
```

Comment:
If you want to I think you can simplify this to just look at rightmost column of the cat output as it shows if it is a write index or not.

Reply:
Yep, I saw that, but it was tricky to keep track of which alias has the missing write index